### PR TITLE
Webchat view home

### PR DESCRIPTION
**Pull request by pedroespindula**
**Feature/Bug description:** Adds home view to webchat.

- On desktop:

homeview

- On mobile with navbar closed:

responsive-closed

- On mobile with navbar open and search input closed:

responsive-open

- On mobile with navbar open and search input open:

responsive-open-search

**TODO/FIXME:** n/a

### CODE REVIEW

File: webchat/components/contacts/sidenav/contacts-sidenav.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<md-sidenav
+        class="md-sidenav-left"
+        md-component-id="left"
+        md-is-locked-open="$mdMedia('gt-md')"
+        md-whiteframe="4"
+        style="height: 100%; width: 100%; display: flex">
```
Comment (JuliePessoa) [inline webchat/components/contacts/sidenav/contacts-sidenav.html]: move style to .css

File: webchat/utils/utils.js [hunk 2]
```diff
@@ -1,6 +1,58 @@
...
 
 const Utils = {
+
+    /**
+     * Identify all the indexes where a substring occurs inside another string
+     * @param {string} substring to be searched
+     * @param {string} fullString that contains the searched substring
+     * @returns {[number]} a list of indexes where the substring occurs
+     */
+    getIndexesOf : function getIndexesOf(substring, fullString) {
```
Comment (JuliePessoa) [inline webchat/utils/utils.js]: test

File: webchat/utils/utils.js [hunk 3]
```diff
@@ -1,6 +1,58 @@
...
+        return indexes;
+    },
+
+    /**
+     * Limit the specified string if its size is bigger than the specified limit
+     * @param {string} stringToLimit string that will be limited if necessary
+     * @param {number} limit max string size
+     * @returns {string} the sliced string, followed by ellipsis, or the original one
+     */
+    limitString : function limitString(stringToLimit, limit) {
```
Comment (JuliePessoa) [inline webchat/utils/utils.js]: test

File: webchat/services/WebchatService.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+(function () {
+    'use strict';
+
+    angular.module("webchat").service('WebchatService',function WebchatService() {
+         const WebchatService = this;
+
+         WebchatService.getContacts = () => [
+             {name: "Name", description: "Description", avatar: "https://www.w3schools.com/howto/img_avatar.png"},
```
Comment (Luiz-FS) [inline webchat/services/WebchatService.js]: Use for loop to generate mocks.

File: webchat/components/contacts/list/contacts-list.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<md-list class="no-margin no-padding custom-scrollbar custom-scrollbar contacts-list-container fill-height">
+    <md-list-item aria-label="user-contact" ng-repeat="user in contactsListCtrl.contacts | filter : contactsListCtrl.searchQuery | orderBy : ['name', 'description']" ng-click="contactsListCtrl.openChat(user)">
```
Comment (brobsc) [inline webchat/components/contacts/list/contacts-list.html]: Add some kind of feedback when there are no online users

File: webchat/test/specs/utils/utils.test.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+(describe('Test utils', () => {
+
+    describe('getIndexesOf()', function () {
+
+        it('should return the indexes where the substring X appear', function () {
```
Comment (JuliePessoa) [inline webchat/test/specs/utils/utils.test.js]: test when theres no substring match

**Pull request merged**